From e1ee4dfd45b89f54f076ab983170487fa3ff0bca Mon Sep 17 00:00:00 2001
From: Laurent Pinchart <laurent.pinchart+renesas@ideasonboard.com>
Date: Mon, 17 Jun 2013 10:37:37 +0900
Subject: [PATCH 493/715] drm: rcar-du: Use the mode active size in mode set

The driver currently configures the dislay to the frame buffer size
while it should use the requested mode active size. Fix it.

Signed-off-by: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
(cherry picked from commit 52b5711ce9077243424f00016f12f9f7df1c8151)

Signed-off-by: Ryo Kataoka <ryo.kataoka.wt@renesas.com>
---
 drivers/gpu/drm/rcar-du/rcar_du_crtc.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/rcar-du/rcar_du_crtc.c b/drivers/gpu/drm/rcar-du/rcar_du_crtc.c
index 1b49b54..08a69db 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_crtc.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_crtc.c
@@ -412,8 +412,8 @@ static int rcar_du_crtc_mode_set(struct drm_crtc *crtc,
 
 	rcrtc->plane->src_x = x;
 	rcrtc->plane->src_y = y;
-	rcrtc->plane->width = crtc->fb->width;
-	rcrtc->plane->height = crtc->fb->height;
+	rcrtc->plane->width = mode->hdisplay;
+	rcrtc->plane->height = mode->vdisplay;
 
 	rcar_du_plane_compute_base(rcrtc->plane, crtc->fb);
 
-- 
1.7.10.4

